src/selectors/entities/groups.js

```diff
@@ -51,6 +51,11 @@ function getTeamGroupIDSet(state, teamID) {
     return new Set(arr);
 }
 
+function getChannelGroupIDSet(state, channelID) {
```

Comment:
this selector will every single time generate a new object, we could create a function for the array, and a selector (based on that function) for the set, that way, if the underneath array hasn't change, the set will be cached.

Reply:
Good point.